extensions/panache/mongodb-panache-common/runtime/src/main/java/io/quarkus/mongodb/panache/common/reactive/runtime/ReactiveMongoOperations.java

```diff
@@ -216,8 +217,20 @@ public Uni<Void> delete(Object entity) {
     public ReactiveMongoCollection mongoCollection(Class<?> entityClass) {
         MongoEntity mongoEntity = entityClass.getAnnotation(MongoEntity.class);
         ReactiveMongoDatabase database = mongoDatabase(mongoEntity);
-        if (mongoEntity != null && !mongoEntity.collection().isEmpty()) {
-            return database.getCollection(mongoEntity.collection(), entityClass);
+        if (mongoEntity != null) {
+            ReactiveMongoCollection collection = mongoEntity.collection().isEmpty()
+                    ? database.getCollection(entityClass.getSimpleName(), entityClass)
+                    : database.getCollection(mongoEntity.collection(), entityClass);
+            if (!mongoEntity.readPreference().isEmpty()) {
+                try {
+                    collection = collection.withReadPreference(ReadPreference.valueOf(mongoEntity.readPreference()));
+                } catch (IllegalArgumentException iae) {
+                    throw new IllegalArgumentException("Illegal read preference " + mongoEntity.readPreference()
```

Comment:
We should probably pass the original exception as the `cause`

Reply:
I agree this is usually a good practice but here it will not give additional information.